types/coin.go

```diff
@@ -863,35 +863,25 @@ func (coins Coins) Sort() Coins {
 
 var (
 	// Denominations can be 3 ~ 128 characters long and support letters, followed by either
-	// a letter, a number or a separator ('/', ':', '.', '_' or '-').
-	reDnmString = `[a-zA-Z][a-zA-Z0-9/:._-]{2,127}`
+	// a letter, a number or a separator ('/', ':', '.', '_' or '-'), or an emoji.
+	reDnmString = fmt.Sprintf(`([a-zA-Z]|%s)([a-zA-Z0-9/:._-]|%s){1,127}`, reEmoji, reEmoji)
+	reEmoji     = `\x{00a9}|\x{00ae}|[\x{2000}-\x{3300}]|\x{d83c}[\x{d000}-\x{dfff}]|\x{d83d}[\x{d000}-\x{dfff}]|\x{d83e}[\x{d000}-\x{dfff}]`
```

Comment:
I'm not happy about this whitelist regex, it's ugly and doesn't even capture all emojis. But going too permissive (i.e. all unicodes?) also seems dangerous.

I think that's the downside of this immutable regex approach: we'll need to find a regex that makes everyone happy.

Reply:
yeah this ain't pretty lol. Why do we need to change the regex again? Just to support ASCII and emojis?